state/prune.go

```diff
@@ -64,6 +64,18 @@ func pruneCollectionAndChildren(mb modelBackend, maxHistoryTime time.Duration, m
 	if err := p.pruneByAge(); err != nil {
 		return errors.Trace(err)
 	}
+	// First try pruning, excluding any items that
+	// have an age field that is not yet set.
+	// ie only prune completed items.
+	if err := p.pruneBySize(); err != nil {
```

Comment:
I feel like this would have been cleaner if you passed in the ageField rather than having it as an attribute of 'p'. The fact that you have to mutate the value of p.ageField and call a function twice expecting different side effects seems like it would be better as a parameter.